src/NLog/Config/ConfigSectionHandler.cs

```diff
@@ -57,12 +57,13 @@ object Create(XmlNode section, IAppDomain appDomain)
             }
             catch (Exception exception)
             {
+                InternalLogger.Error(exception, "ConfigSectionHandler error.");
+              
                 if (exception.MustBeRethrown())
                 {
                     throw;
                 }
-
-                InternalLogger.Error("ConfigSectionHandler error: {0}", exception);
+                
                 throw;
```

Comment:
We rethrow the exception in any case (and we log explicitly log it). Isn't the `if` above redundant?